src/notation/internal/notationinteraction.cpp

```diff
@@ -7339,6 +7339,57 @@ void NotationInteraction::addMelisma()
     toLyrics->selectAll(toLyrics->cursor());
 }
 
+void NotationInteraction::addElision() {
+
+    if (!m_editData.element || !m_editData.element->isLyrics()) {
+        LOGW("addElision called outside of lyrics editing context.");
+        return;
+    }
+
+    // fetch the text‐edit data & cursor
+    auto edd = m_editData.getData(m_editData.element);
+    if (!edd) {
+        LOGE("addElision: Failed to get EditData for the element.");
+        return;
+    }
+    auto* ted = static_cast<TextEditData*>(edd.get());
+    if (!ted) {
+        LOGE("addElision: EditData is not TextEditData.");
+        return;
+    }
+
+    TextCursor* cursor = ted->cursor();
+    if (!cursor) {
+        LOGE("addElision: Failed to get TextCursor from TextEditData.");
+        return;
+    }
+
+    // build the elision symbol and set the engraving font
+    String elisionStr = score()->engravingFont()->toString(mu::engraving::SymId::lyricsElision);
+    cursor->format()->setFontFamily(u"ScoreText");
+
+    // insert as one undoable action
+    score()->undo(new InsertText(cursor, elisionStr), &m_editData);
```

Comment:
One more thing, instead of this, it might be nicer to make `TextBase::insertSym` public and just call that (`toTextBase(m_editData.element)->insertSym(…)`). The `insertText` method is public too, and `insertSym` isn't really different from that one, so I think it's justifiable to make it public too.